File: client/src/logger.ts
/**
 * Minimalist logger with ISO timestamps
 * Wraps native console API with timestamp prefixes
 * Follows AGENTS.md: strict typing, zero dependencies, minimal code
 */

function getTimestamp(): string {
  return new Date().toISOString();
}

export const logger = {
  debug: (...args: unknown[]): void => {
    console.debug(`[${getTimestamp()}]`, ...args);
  },
  info: (...args: unknown[]): void => {
    console.info(`[${getTimestamp()}]`, ...args);
  },
  warn: (...args: unknown[]): void => {
    console.warn(`[${getTimestamp()}]`, ...args);
  },
  error: (...args: unknown[]): void => {
    console.error(`[${getTimestamp()}]`, ...args);
  },
  log: (...args: unknown[]): void => {
    console.log(`[${getTimestamp()}]`, ...args);
  }
};

File: server/src/main/scala/mg2/Logger.scala
package mg2

import java.time.Instant
import java.time.ZoneId
import java.time.format.DateTimeFormatter

/**
 * Minimalist logger with ISO timestamps for Scala
 * Wraps println with timestamp prefixes
 * Follows AGENTS.md: strict typing, zero dependencies, minimal code
 */
object Logger {
  private val formatter = DateTimeFormatter.ISO_INSTANT.withZone(ZoneId.systemDefault())

  private def getTimestamp(): String = {
    formatter.format(Instant.now())
  }

  def debug(msg: String): Unit = {
    println(s"[${getTimestamp()}] $msg")
  }

  def info(msg: String): Unit = {
    println(s"[${getTimestamp()}] $msg")
  }

  def warn(msg: String): Unit = {
    println(s"[${getTimestamp()}] $msg")
  }

  def error(msg: String): Unit = {
    println(s"[${getTimestamp()}] $msg")
  }
}
